src/TransactionProgressLabel.jsx

```diff
@@ -1,46 +1,61 @@
 const React = require('react');
-const {ReactiveComponent} = require('oo7-react');
-const {Label, Icon} = require('semantic-ui-react');
+const { ReactiveComponent } = require('oo7-react');
+const { Label, Icon } = require('semantic-ui-react');
 
 function styleStatus (value) {
 	return (
-		value.initialising !== undefined || value.estimating !== undefined ? { text: 'estimating', icon: 'bullseye', color: 'blue', basic: true } :
-		value.estimated ||
-		value.requested ? { text: 'authorising', icon: 'key', color: 'orange', basic: true } :
-		value.scheduled ? { text: 'scheduled', icon: 'spinner', color: 'green', basic: false, loading: true } :
-		value.signed ? { text: 'finalising', icon: 'spinner', color: 'green', basic: true, loading: true } :
-		value.confirmed ? { text: 'finalised', icon: 'check', color: 'green', basic: false } :
-		value.failed ?
-			value.failed.code == -32015 ? { text: 'invalid', icon: 'exclamation', color: 'red', basic: true } :
-			value.failed.code == -32040 ? { text: 'rejected', icon: 'x', color: 'grey', basic: true } :
-			null :
-		null
+		value.initialising !== undefined || value.estimating !== undefined
+			? { text: 'estimating', icon: 'bullseye', color: 'blue', basic: true }
+			: value.estimated || value.requested
+				? { text: 'authorising', icon: 'key', color: 'orange', basic: true }
+				: value.scheduled
+					? { text: 'scheduled', icon: 'spinner', color: 'green', basic: false, loading: true }
+					: value.signed
+						? { text: 'finalising', icon: 'spinner', color: 'green', basic: true, loading: true }
+						: value.confirmed
```

Comment:
This block is quite unreadable. Use switch (better) or if/elseif (if switch not possible)

Reply:
fixed :)